hw/opentitan/ot_spi_device.c

```diff
@@ -2498,10 +2510,12 @@ static void ot_spi_device_tpm_addr_state(OtSPIDeviceState *s,
     }
 
     tpm->can_receive = 1u;
-    tpm->should_sw_handle =
-        ot_spi_device_is_tpm_mode_crb(s) ||
-        ot_spi_device_tpm_disable_hw_regs(s) || buf[0u] != TPM_ADDR_HEADER ||
-        !ot_spi_device_tpm_is_hw_register(tpm->reg);
+    tpm->should_sw_handle = true;
+    if (ot_spi_device_is_tpm_mode_crb(s) &&
```

Comment:
Is this correct? It suggests that we only handle in HW if the TPM is in CRB mode, but I think the correct logic is supposed to express that the TPM should be in FIFO mode?


```suggestion
    if (!ot_spi_device_is_tpm_mode_crb(s) &&
```